core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -251,7 +251,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     private void failWithUnexpectedMessageType(ChannelHandlerContext ctx, Object msg, Class<?> expected) {
         fail(ctx, new ProtocolViolationException(
                 "unexpected message type: " + msg.getClass().getName() +
-                " (expected: " + expected.getName() + ')'));
+                " (expected: " + expected.getName() + ", channel: " + ctx.channel() +
+                ", resId: " + resId + ", lastPingReqId: " + lastPingReqId + ')'));
```

Comment:
How about adding `lastPingReqId` only if `lastPingReqId` is not equal to -1

Reply:
Fixed. 😉